agent/php_user_instrument.c

```diff
@@ -644,6 +576,18 @@ void nr_php_user_function_add_declared_callback(const char* namestr,
   }
 }
 
+#if ZEND_MODULE_API_NO >= ZEND_7_4_X_API_NO
+nruserfn_t* nr_php_get_wraprec(zend_function* zf) {
+  nruserfn_t* wraprec = NULL;
+  /* Look in user function wrappers first: */
+  wraprec = nr_php_wraprec_hashmap_get(user_function_wrappers, zf);
+  if (NULL == wraprec) {
+    /* Look in transient user function wrappers next: */
+    wraprec = nr_php_wraprec_hashmap_get(transient_wrappers, zf);
+  }
+  return wraprec;
+}
```

Comment:
Is there a way we could put all wraprecs in a single hash and remote the transients at the end of a request?  It seems for the case where the number of transient wraprecs is large (>100?) this is going to incur an additional hash lookup for each one and we know there can be a large number.

Reply:
It is possible to put all wraprecs in a single hashmap. This would require a slight change to hashmap destruction at the end of the request.

Reply:
Here's how it could be done: https://github.com/misiekn/newrelic-php-agent/commit/c82ba80c2030aac92de5881fdb55a24a6340065e.

Reply:
Your new addition looks good Michal.